### PR TITLE
[SPARK-12632][Python][Make Parameter Descriptions Consistent for PySpark MLlib FPM and Recommendation]

### PR DESCRIPTION
**Pull request by somideshmukh**
Made changes in FPM file ,Recommendation file doesnot contain param changes

### CODE REVIEW

File: python/pyspark/mllib/fpm.py [hunk 1]
```diff
@@ -68,11 +68,14 @@ def train(cls, data, minSupport=0.3, numPartitions=-1):
...
-            transaction.
-        :param minSupport: The minimal support level (default: `0.3`).
-        :param numPartitions: The number of partitions used by
-            parallel FP-growth (default: same as input data).
+         :param data:
+           The input data set, each element contains a transaction.
+         :param minSupport:
+           The minimal support level
+           (default: `0.3`)
+         :param numPartitions:The number of partitions used by parallel FP-growth
```
Comment (vijaykiran) [inline python/pyspark/mllib/fpm.py]: You missed this one :)

File: python/pyspark/mllib/fpm.py [hunk 2]
```diff
@@ -130,15 +133,22 @@ def train(cls, data, minSupport=0.1, maxPatternLength=10, maxLocalProjDBSize=320
...
-        :param maxLocalProjDBSize: The maximum number of items (including delimiters used in
-            the internal storage format) allowed in a projected database before local
-            processing. If a projected database exceeds this size, another
-            iteration of distributed prefix growth is run. (default: `32000000`)
+        :param data:
+          The input data set, each element contains a sequnce of itemsets.
+        :param minSupport:
+          The minimal support level of the sequential pattern, any pattern appears
+          more than  (minSupport * size-of-the-dataset) times will be output.
+          default: `0.1`)
```
Comment (vijaykiran) [inline python/pyspark/mllib/fpm.py]: I think the format should be (default: `0.1`).

Comment (BryanCutler) [inline python/pyspark/mllib/fpm.py]: yes, please use the format that @vijaykiran has above (although, no period after the default)

File: python/pyspark/mllib/fpm.py [hunk 3]
```diff
@@ -68,11 +68,14 @@ def train(cls, data, minSupport=0.3, numPartitions=-1):
...
 
-        :param data: The input data set, each element contains a
-            transaction.
-        :param minSupport: The minimal support level (default: `0.3`).
-        :param numPartitions: The number of partitions used by
-            parallel FP-growth (default: same as input data).
+         :param data:
+           The input data set, each element contains a transaction.
+         :param minSupport:
+           The minimal support level
```
Comment (BryanCutler) [inline python/pyspark/mllib/fpm.py]: Please add a period after the end of the description.

File: python/pyspark/mllib/fpm.py [hunk 4]
```diff
@@ -130,15 +133,22 @@ def train(cls, data, minSupport=0.1, maxPatternLength=10, maxLocalProjDBSize=320
...
+          default: `0.1`)
+        :param maxPatternLength:
+          The maximal length of the sequential pattern, any pattern appears
+          less than maxPatternLength will be output.
+          (default: `10`)
+        :param maxLocalProjDBSize:
+          The maximum number of items (including delimiters used in
+          the internal storage format) allowed in a projected database before local
+          processing. If a projected database exceeds this size, another
+          iteration of distributed prefix growth is run.
```
Comment (BryanCutler) [inline python/pyspark/mllib/fpm.py]: Line length can go up to 100 characters, can you please combine these lines and other multi-line descriptions to that limit?

File: python/pyspark/mllib/fpm.py [hunk 5]
```diff
@@ -68,11 +68,14 @@ def train(cls, data, minSupport=0.3, numPartitions=-1):
         """
         Computes an FP-Growth model that contains frequent itemsets.
 
-        :param data: The input data set, each element contains a
-            transaction.
-        :param minSupport: The minimal support level (default: `0.3`).
-        :param numPartitions: The number of partitions used by
-            parallel FP-growth (default: same as input data).
+         :param data:
```
Comment (BryanCutler) [inline python/pyspark/mllib/fpm.py]: There are extra spaces in the indentations here.  :param should line up with 'Computes' as before

**Comment by BryanCutler** [comment]
Thanks @somideshmukh!  Could you please look the corrections from @vijaykiran and me, extend the descriptions to the 100 character limit, and add the parameter descriptions to recommendation.py?

File: python/pyspark/mllib/fpm.py [hunk 6]
```diff
@@ -67,12 +67,15 @@ class FPGrowth(object):
     def train(cls, data, minSupport=0.3, numPartitions=-1):
         """
         Computes an FP-Growth model that contains frequent itemsets.
+         :param data:
+           The input data set, each element contains a transaction.
+         :param minSupport:
+           The minimal support level.
+           (default: 0.3)
+         :param numPartitions:
+		   The number of partitions used by parallel FP-growth.
```
Comment (vijaykiran) [inline python/pyspark/mllib/fpm.py]: too much indentation here.

File: python/pyspark/mllib/fpm.py [hunk 7]
```diff
@@ -130,15 +133,21 @@ def train(cls, data, minSupport=0.1, maxPatternLength=10, maxLocalProjDBSize=320
...
-        :param maxPatternLength: the maximal length of the sequential pattern, any pattern appears
-            less than maxPatternLength will be output. (default: `10`)
-        :param maxLocalProjDBSize: The maximum number of items (including delimiters used in
-            the internal storage format) allowed in a projected database before local
-            processing. If a projected database exceeds this size, another
-            iteration of distributed prefix growth is run. (default: `32000000`)
+        :param data:
+          The input data set, each element contains a sequnce of itemsets.
+        :param minSupport:
+          The minimal support level of the sequential pattern, any pattern appears more than
```
Comment (thunterdb) [inline python/pyspark/mllib/fpm.py]: the lines below have indentation issues

File: python/pyspark/mllib/recommendation.py [hunk 8]
```diff
@@ -239,6 +239,17 @@ def train(cls, ratings, rank, iterations=5, lambda_=0.01, blocks=-1, nonnegative
         product of two lower-rank matrices of a given rank (number of features). To solve for these
         features, we run a given number of iterations of ALS. This is done using a level of
         parallelism given by `blocks`.
+		
+		:param iterations:
```
Comment (thunterdb) [inline python/pyspark/mllib/recommendation.py]: indentation issues?

**Comment by jkbradley** [comment]
I just added a note to the parent JIRA about a formatting issue affecting all 5 PRs: [https://issues.apache.org/jira/browse/SPARK-11219?focusedCommentId=15090225&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-15090225]
Could you please check it out & ping when I should review again?  Thank you!

File: python/pyspark/mllib/fpm.py [hunk 9]
```diff
@@ -67,12 +67,14 @@ class FPGrowth(object):
...
-        :param minSupport: The minimal support level (default: `0.3`).
-        :param numPartitions: The number of partitions used by
-            parallel FP-growth (default: same as input data).
+        :param data:
+          The input data set, each element contains a transaction.
+        :param minSupport:
+          The minimal support level.
+          (default: 0.3)
+        :param numPartitions:
+		  The number of partitions used by parallel FP-growth.
```
Comment (BryanCutler) [inline python/pyspark/mllib/fpm.py]: Too much indentation here

File: python/pyspark/mllib/fpm.py [hunk 10]
```diff
@@ -128,17 +130,25 @@ class PrefixSpan(object):
...
-        :param minSupport: the minimal support level of the sequential pattern, any pattern appears
-            more than  (minSupport * size-of-the-dataset) times will be output (default: `0.1`)
-        :param maxPatternLength: the maximal length of the sequential pattern, any pattern appears
-            less than maxPatternLength will be output. (default: `10`)
-        :param maxLocalProjDBSize: The maximum number of items (including delimiters used in
-            the internal storage format) allowed in a projected database before local
-            processing. If a projected database exceeds this size, another
-            iteration of distributed prefix growth is run. (default: `32000000`)
+        Finds the complete set of frequent sequential patterns in the input
+		sequences of itemsets.
```
Comment (BryanCutler) [inline python/pyspark/mllib/fpm.py]: this should have extra indentation, it should line up with the word "Finds"

File: python/pyspark/mllib/fpm.py [hunk 11]
```diff
@@ -128,17 +130,25 @@ class PrefixSpan(object):
...
+        :param data:
+          The input data set, each element contains a sequnce of itemsets.
+        :param minSupport:
+          The minimal support level of the sequential pattern, any pattern
+		  appears more than (minSupport * size-of-the-dataset) times will be
```
Comment (BryanCutler) [inline python/pyspark/mllib/fpm.py]: Same as above, no extra indentation when the line wraps around

**Comment by somideshmukh** [comment]
We have made  same changes and committed the code,but when pull request is created ,it i showing indentation problem.I am attaching image  will show changes  we have made.

recommendation-changes-comparision

fpm-comparision

**Comment by BryanCutler** [comment]
Hi @somideshmukh , it looks like your changes still have a couple of indentation problems.  Maybe try changing the indent to not use the tab character.  If using Intellij, you can set this from File -> Settings, then under Editor -> Code Style, for "Default Indent Options" uncheck the box that says "Use tab character" and set "Tab size" to 2, "Indent" to 2" and "Continuation indent" to 4.

**Comment by BryanCutler** [comment]
Hi @somideshmukh , any update for this?  Let me know if you need any assistance.

**Comment by BryanCutler** [comment]
Thanks for working on this @somideshmukh , I finished up the remaining work in https://github.com/apache/spark/pull/11186.  Could you please close this PR and the other?

cc @mengxr

**Pull request closed without merging**